User.ts

```diff
@@ -0,0 +1,91 @@
+// Base interface
+interface BaseUser {
+  name: string;
+  email: string;
+}
+
+
+interface Admin extends BaseUser {
+  createUser(user: User): void;
+}
+
+
+interface Moderator extends BaseUser {
+  banUser(user: User): void;
+}
+
+interface RegularUser extends BaseUser {
+  viewContent(): void;
+}
+type User = Admin | Moderator | RegularUser;
+function isAdmin(user: User): user is Admin {
+  return typeof (user as Admin).createUser === 'function';
```

Comment:
you dont need user as admin here since you already defined that user is Admin in the function return type

Reply:
Same comment for the stuff below